File: animals.cpp
/*
* Filename: animals.cpp
* Project: Assignment 2
* By: Jongeon Lee
* Student number: 8790144
* Date: Oct 14, 2023
* Description: A CGI script that displays information about a selected animal based on user input.
*/

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <direct.h>
#pragma warning(disable : 4996)

using namespace std;

int main() {
    // Get query parameters from the URL
    char* query_string = getenv("QUERY_STRING");

    // Initialize variables to store user input
    char* name = NULL;
    char* animal = NULL;

    // Parsing the name and animals
    char* token = strtok(query_string, "&");

    while (token != NULL) {
        char* equal_sign = strchr(token, '=');
        if (equal_sign != NULL) {
            char* value = equal_sign + 1;
            if (strncmp(token, "name", equal_sign - token) == 0) {
                name = value;
            }
            else if (strncmp(token, "animal", equal_sign - token) == 0) {
                animal = value;
            }
        }
        token = strtok(NULL, "&");
    }

    // Output the HTTP header
    printf("Content-type: text/html\n\n");

    // Start the HTML content
    printf("<html>\n");
    printf("<head>\n");
    printf("<link rel=\"stylesheet\" href=\"style.css\">\n");
    printf("<title>Animal Serve in CGI</title>\n");
    printf("</head>\n");
    printf("<body>\n");
    printf("<div class=\"headerArea\">Pet Facts!</div>\n");
    printf("<div class=\"division-line\"></div>\n");
    printf("<div class=\"greetingArea\">\n");
    printf("Hello, %s! You selected to learn more about the %s.\n", name, animal);
    printf("<br>Look how amazing they can be!\n");
    printf("</div>\n");
    printf("<div class=\"division-line\"></div>\n");
    printf("<div class=\"imageArea\">\n");
    printf("<img src=\"theZoo/%s.png\" alt=\"image\">\n", animal);
    printf("</div>\n");

    // TXT file open and read function
    char filePath[256] = { NULL };
    snprintf(filePath, sizeof(filePath), "theZoo/%s.txt", animal);

    FILE* fp = fopen(filePath, "r");

    // Error check
    char line[256] = { '\0' };
    if (fp == NULL)
    {
        printf("Error-fopen function: Can't open file for reading\n");
        return -1;
    }

    // Read file contents
    int count = 1;
    printf("<div class=\"description-line\">");
    while (fgets(line, sizeof(line), fp)) {

        if (count == 1) {
            printf("<div class=\"description-line-first\" style='font-size: 22px;'>%s</div>\n", line);
        }
        else if (count == 2) {
            printf("<div class=\"description-line-second\" style='font-size: 25px;'>%s</div>\n", line);
        }
        else {
            printf("<div class=\"description-line-third\" style='font-size: 18px;'>%s</div>\n", line);
        }
        count++;
    }
    printf("</div>\n");

    printf("</body>\n</html>");

    // File close
    fclose(fp);
    return 0;
}

File: animals.php
<!--
 Filename: animals.php
 Project: Assignment 2
 By: Jongeon Lee
 Student number: 8790144
 Date: Oct 14, 2023
 Description: A PHP script that displays information about a selected animal based on user input.
-->

<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <link rel="stylesheet" href="style.css">
    <title>Animal Serve in PHP</title>
</head>
<body>
    <?php

    // Initialize variables to store user input
    $name = '';
    $animal = '';

    // Get query parameters from the URL
    $query_string = $_SERVER['QUERY_STRING'];
    parse_str($query_string, $params);

    // Retrieve the user's name and selected animal if provided
    if (isset($params['name'])) {
        $name = $params['name'];
    }

    if (isset($params['animal'])) {
        $animal = $params['animal'];
    }

    // Set the page title dynamically
    echo "<script>document.title = 'Animal Serve in PHP';</script>";

    // Display header and greeting
    echo "<div class=\"headerArea\">Pet Facts!</div>\n";
    echo "<div class=\"division-line\"></div>\n";
    echo "<div class=\"greetingArea\">\n";
    echo "Hello, $name! You selected to learn more about the $animal.\n";
    echo "<br>Look how amazing they can be!\n";
    echo "</div>\n";
    echo "<div class=\"division-line\"></div>\n";
    echo "<div class=\"imageArea\">\n";
    echo "<img src=\"theZoo/$animal.png\" alt=\"image\">\n";
    echo "</div>\n";

    // Read and display information from a text file
    $filePath = "theZoo/$animal.txt";
    $fileContents = file($filePath);

    if ($fileContents === false) {
        echo "Error: Can't open the file for reading.";
    } else {
        $count = 1;
        echo "<div class=\"description-line\">";
        foreach ($fileContents as $line) {
            if ($count == 1) {
                echo "<div class=\"description-line-first\" style='font-size: 22px;'>$line</div>\n";
            } else if ($count == 2) {
                echo "<div class=\"description-line-second\" style='font-size: 25px;'>$line</div>\n";
            } else {
                echo "<div class=\"description-line-third\" style='font-size: 18px;'>$line</div>\n";
            }
            $count++;
        }
        echo "</div>\n";
    }

    echo "</body>\n</html>";
    ?>

</body>
</html>
